src/main/java/com/autotune/analyzer/recommendations/autoscaler/instaslice/InstasliceHelper.java

```diff
@@ -0,0 +1,133 @@
+package com.autotune.analyzer.recommendations.autoscaler.instaslice;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import io.fabric8.kubernetes.api.model.GenericKubernetesResource;
+import io.fabric8.kubernetes.client.DefaultKubernetesClient;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.fabric8.kubernetes.client.dsl.base.CustomResourceDefinitionContext;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicReference;
+
+public class InstasliceHelper {
+
+    private static InstasliceHelper instance;
+    private static final String instasliceNamespace = "instaslice-system";
+    private static final String group = "inference.redhat.com";
+    private static final String version = "v1alpha1";
+    private static final String namespace = "instaslice-system";
+    private static final String plural = "instaslices";
+    private static final String scope = "Namespaced";
+
+
+    private InstasliceHelper() {
+    }
+
+    private boolean checkIfInstasliceExist() {
+        try (KubernetesClient client = new DefaultKubernetesClient()){
+            return client.namespaces().withName(namespace).get() != null;
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return false;
+    }
+
+    private List<GenericKubernetesResource> getInstasliceObjects() {
+        try (KubernetesClient client = new DefaultKubernetesClient()) {
+            CustomResourceDefinitionContext crdContext = new CustomResourceDefinitionContext.Builder()
+                    .withGroup(group)
+                    .withVersion(version)
+                    .withScope(scope)
+                    .withPlural(plural)
+                    .build();
+
+            return client.genericKubernetesResources(crdContext).inNamespace(namespace).list().getItems();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+        return null;
+    }
+
+    public static InstasliceHelper getInstance() {
+        if (null == instance) {
+            synchronized (InstasliceHelper.class) {
+                if (null == instance) {
+                    instance = new InstasliceHelper();
+                }
+            }
+        }
+        return instance;
+    }
+
+    public String getUUID(String workloadNamespace, String workloadName) {
+        if (!checkIfInstasliceExist())
+            return null;
+
+        AtomicReference<String> uuid = new AtomicReference<>(null);
+
+        List<GenericKubernetesResource> instasliceObjects = getInstasliceObjects();
+        ObjectMapper mapper = new ObjectMapper();
+
+        if (null != instasliceObjects && !instasliceObjects.isEmpty()) {
+            try {
+                instasliceObjects.forEach(item -> {
+                    String name = item.getMetadata().getName();
+                    System.out.println("Found Instaslice: " + name);
```

Comment:
Please use LOGGER.info where absolutely essential and LOGGER.debug in the rest of the places

Reply:
Updated print statements to Logger.